test/registry/withdraw.js

```diff
@@ -42,23 +42,77 @@ contract('Registry', (accounts) => {
       const registry = await Registry.deployed();
       const listing = utils.getListingHash('shouldntwithdraw.net');
 
+      const deposit = minDeposit.plus(bigTen(1));
+
       // Whitelist, then challenge
-      await utils.addToWhitelist(listing, minDeposit, applicant);
+      await utils.addToWhitelist(listing, deposit, applicant);
       await utils.as(challenger, registry.challenge, listing, '');
 
       try {
         // Attempt to withdraw; should fail
-        await utils.as(applicant, registry.withdraw, listing, withdrawAmount);
-        assert.strictEqual(false, 'Applicant should not have been able to withdraw from a challenged, locked listing');
+        await utils.as(applicant, registry.withdraw, listing, '1');
       } catch (err) {
-        const errMsg = err.toString();
-        assert(utils.isEVMException(err), errMsg);
+        assert(utils.isEVMException(err), err.toString());
+        return;
       }
+      assert.strictEqual(false, 'Applicant should not have been able to withdraw from a challenged, locked listing');
       // TODO: check balance
       // TODO: apply, gets challenged, and then minDeposit lowers during challenge.
       // still shouldn't be able to withdraw anything.
       // when challenge ends, should be able to withdraw origDeposit - new minDeposit
     });
+
+    it('should revert if the message sender is not the owner of the application/listing', async () => {
+      const registry = await Registry.deployed();
+      const listing = utils.getListingHash('challengerWithdraw.net');
+
+      const deposit = minDeposit.plus(bigTen(1));
+
+      // Whitelist
+      await utils.addToWhitelist(listing, deposit, applicant);
+
+      try {
+        // Attempt to withdraw; should fail
+        await utils.as(challenger, registry.withdraw, listing, '1');
+      } catch (err) {
+        assert(utils.isEVMException(err), err.toString());
+        return;
+      }
+      assert(false, 'non-owner should not be able to withdraw from listing.');
+    });
+
+    it('should allow listing owner to withdraw and decrease the UnstakedDeposit', async () => {
+      const registry = await Registry.deployed();
+      const listing = utils.getListingHash('ITWORKS.net');
+
+      const deposit = minDeposit.plus(bigTen(1));
+
+      // Whitelist
+      await utils.addToWhitelist(listing, deposit, applicant);
+
+      await utils.as(applicant, registry.withdraw, listing, '1');
+
+      const afterWithdrawDeposit = await utils.getUnstakedDeposit(listing);
+
+      assert.strictEqual('10', afterWithdrawDeposit, 'UnstakedDeposit should be 0.');
+    });
+
+    it('should not allow withdrawal greater than UnstakedDeposit', async () => {
+      const registry = await Registry.deployed();
+      const listing = utils.getListingHash('moreThanIOwn.net');
+
+      const withdrawGreaterAmount = minDeposit.plus(bigTen(1));
+
+      // Whitelist
+      await utils.addToWhitelist(listing, minDeposit, applicant);
```

Comment:
Get minDeposit from parameterizer.
Get unstakedDeposit to calculate how much to withdraw.